### ISSUE TITLE
Does prefer-constant really have to trigger on 'return true;'?

### ISSUE THREAD

**Issue by Maruz** [wix/eslint-plugin-lodash#101]
I am working with React.js, and often just use `return true;` in various component functions. I've been experimenting with this ESlint plugin and noticed that when I turn on the `lodash/prefer-constant` rule, the linter wants me to use the `_.constant()` function in these cases. 

To make this lint rule pass, I then have to change `return true;` to `return _.constant(true);` which does not feel very sane. Should not this rule just apply to potentially mutable values? Numbers / booleans (and other natively immutable values) should perhaps be ignored by this rule?

**Comment by ganimomer** [comment]
Hi, I think there's a slight mixup here about what this rule warns against.
`_.constant(true)` doesn't return `true`, it returns a **function** that returns true.
Say you have a function that returns a value:
```js
const f = function() {return true};
```
This can be changed to:
```js
const f = _.constant(true);
```
This is actually *more* suited for primitives, since passing a mutable value makes it shared along all function calls:
```js
const getEmptyArr = _.constant([]);
const a = getEmptyArr();
const b = getEmptyArr();
a.push(1);
console.log(b) //[1]
```

**Comment by Maruz** [comment]
That may be so, but this rule triggers if I write this:

`import React, { Component } from 'react';


export default class App extends Component {

  shouldComponentUpdate() {
    return true;
  }

  render() {
    return <div />;
  }

}
`

`6:24  error  Prefer _.constant over a function returning a literal  lodash/prefer-constant`

**Comment by jfmengels** [comment]
In this case, the rule expects you to write

```js
export default class App extends Component {
  shouldComponentUpdate: _.constant(true);
  // ... 
}
```
If you still want to use your version, you're better off disabling the rule for that line, so adding `// eslint-disable-line lodash/prefer-constant` on that line.
